src/Elastic.CommonSchema/Base.Serialization.cs

```diff
@@ -96,16 +97,29 @@ protected internal virtual bool TryRead(string propertyName, out Type type)
 
 		public byte[] SerializeToUtf8Bytes() => JsonSerializer.SerializeToUtf8Bytes(this, GetType(), SerializerOptions);
 
+		[ThreadStatic]
+		private static ReusableUtf8JsonWriter _reusableJsonWriter;
```

Comment:
Must admit I really dislike `[ThreadStatic]`-objects. Why not let ReusableUtf8JsonWriter live in the NLog-project, until others needs it?

Reply:
Not besides the clutter of having the object allocated for every threadpool thread, and the inability to release the object unless the thread is recycled. Anyway not my library to maintain or use.

Reply:
@russcam what would be the benefit? The way its currently implemented runs the initialization on each thread too. `IDisposable` brings no benefit in static context. 

If having an object allocated for each threadpool thread is an issue we can remove `ThreadStatic` alltogether and run with a single instance for simplicity sake knowing cache misses inside the `ReusableUtf8JsonWriter` are more likely.

This to me is a prime example of the few cases I'd want ThreadStatic as the lifetime is explictly bound to the thread and we always want to have 1 instance per thread. Since this is part of logging which is likely happening throughout the whole lifetime of a thread I'm also somewhat less concerned about releasing.

Since this is contentious wIil remove the `ThreadStatic` for now and we can revisit in the future.

Reply:
> what would be the benefit? The way its currently implemented runs the initialization on each thread too. IDisposable brings no benefit in static context.

`ThreadLocal<ReusableUtf8JsonWriter>` brings declaration and initialization together, so maybe a little easier to reason about? Technically they would both work the same here as far as I can see